deepdrivemd/data/stream/OutlierDB.py

```diff
@@ -0,0 +1,77 @@
+import os
+import random
+from typing import List, Tuple
+
+
+class OutlierDB:
+    """Stores the metadata for outliers to be used by simulations.
+
+    Attributes
+    ----------
+    dir : str
+          directory with published outliers
+    sorted_index: List[str]
+          list of md5sums of outlier positions (used as a name of
+          an outlier pdb or numpy file) sorted by the corresponding rmsd
+    dictionary: Dict
+          maps md5sum to rmsd
+    """
+
+    def __init__(self, dir: str, restarts: List[Tuple[float, str]]):
+        """Constructor
+
+        Parameters
+        ----------
+        dir : str
+              directory with published outliers
+        restarts : List[Tuple[float, str]]
+              list of outliers given as tuples of rmsd and md5sum of positions (used as a file name)
+        """
+        self.dir = dir
+        self.sorted_index = list(
+            map(lambda x: os.path.basename(x[1]).replace(".pdb", ""), restarts)
+        )
+        self.dictionary = {}
+        for rmsd, path in restarts:
+            md5 = os.path.basename(path).replace(".pdb", "")
+            self.dictionary[md5] = rmsd
+        self.print()
+
+    def print(self, n: int = 5):
+        print("=" * 30)
+        print("In OutlierDB")
+        n = min(n, len(self.sorted_index))
+        for i in range(n):
+            md5 = self.sorted_index[i]
+            print(f"{md5}: {self.dictionary[md5]}")
+        print("=" * 30)
+
+    def next_random(self, m: int = None) -> str:
+        """Return next outlier using beta distribution that prefers smaller rmsds
+
+        Parameters
+        ----------
+        m : int, default = None
+            if `m` is not `None`, restrict the random selection to the first
+            `m` elements of `softed_index`, otherwise - any element can be chosen.
+
+        """
+        if len(self.sorted_index) == 0:
+            print("Bug")
+            return None
```

Comment:
Should not return None, since it clashes with str return type. Shouldn't this simply raise an exception or at least throw an assert? What does it mean for there to be a bug here. Could we raise a ValueError with an explanation of what is going on? Does a bug here mean that DeepDriveMD should stop?

Reply:
Since it never happened, it did not matter. Changed to raising ValueError